src/mainwindow.cpp

```diff
@@ -110,27 +110,17 @@ bool MainWindow::eventFilter(QObject* /*object*/, QEvent* event)
         const int tabRegion = getTabBar()->height() + getTopWidget()->height() + 30;
         int clickY = portutils::getY(*mouseEvent);
         // We don't have to check for visibilty as calling hide() on a hidden widget, or show() on a non-hidden widget is a no-op
-        if (clickY == 0) {
-            showTabAndTop();
-        } else if(clickY >= tabRegion) {
-            hideTabAndTop();
-        }
+        if (clickY == 0) { showTabAndTop(); } 
+        else if(clickY >= tabRegion) { hideTabAndTop(); }
     }
     return false;
 }
 
 void MainWindow::tabChanged(bool libraryDisplayed) 
 {
     QAction *readingList = KiwixApp::instance()->getAction(KiwixApp::ToggleReadingListAction);
-    if(libraryDisplayed) {
-        if(readingList->isChecked()) readingListRemainActive = true;
-        libraryPageDisplayed();
-    } else if (readingListRemainActive) { // Leaving the library && before the reading list was active
-        readingList->setChecked(true);
-        readingListRemainActive = false;
-    } else if (!readingList->isChecked()) { // Not in library && readinglist inactive -> hide the side bar
-        mp_ui->sideBar->hide();
-    }
+    if(libraryDisplayed) { libraryPageDisplayed(); } 
+    else if (readingList->isChecked()) { mp_ui->sideBar->setCurrentWidget(mp_ui->readinglistbar); }
 }
```

Comment:
Now you can have the sidebar of the library view on content tabs:

![Screenshot from 2024-04-26 12-27-20](https://github.com/kiwix/kiwix-desktop/assets/11933502/ddc589e7-76b7-4cb7-b9dc-b1e1588279bf)

You should hide the sidebar when switching to a content tab while the reading list button is unchecked.

Reply:
Missed this. Fixing on next commit :)